### PR TITLE
Use instruction cache to speed up decoding.

### PR DESCRIPTION
**Pull request by ptersilie**
According to some measurements taken by @vext01 this leads to speed ups of up to 4x.

### CODE REVIEW

File: src/backends/perf_pt/decode.c [hunk 1]
```diff
@@ -153,8 +154,11 @@ perf_pt_init_block_decoder(void *buf, uint64_t len, int vdso_fd, char *vdso_file
         goto clean;
     }
 
+    // Use image cache to speed up decoding.
+    struct pt_image_section_cache *iscache = pt_iscache_alloc(NULL);
```
Comment (ltratt) [inline src/backends/perf_pt/decode.c:158]: I think this value needs to be checked to see if it's `NULL`.

Comment (vext01) [reply src/backends/perf_pt/decode.c:158]: Agreed:

> Returns a new traced memory image section cache on success, NULL otherwise.

Comment (ptersilie) [reply src/backends/perf_pt/decode.c:158]: Fixed in [75616b9](https://github.com/ykjit/hwtracer/pull/98/commits/75616b925502e6fa1bc48e8de6b864086d3d1626)

**Comment by ptersilie** [comment]
Ready to squash?

**Comment by ltratt** [comment]
Please squash.

**Comment by ptersilie** [comment]
Squashed.

**Pull request merged**